modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala

```diff
@@ -74,48 +168,117 @@ object GlobalL0Service {
+              .handleErrorWith { e =>
+                logger
+                  .warn(e)(s"Failure pulling snapshot with ordinal=$ordinal")
+                  .map(_ => snapshots.asRight[(List[SnapshotOrdinal], List[Hashed[GlobalIncrementalSnapshot]])])
+              }
+
+          case (Nil, snapshots) =>
+            Applicative[F].pure(snapshots.asRight[(List[SnapshotOrdinal], List[Hashed[GlobalIncrementalSnapshot]])])
+        }
+      }
+
+      private case class MajoritySnapshotData(ordinal: SnapshotOrdinal, hash: Hash)
+
+      private def getLatestMajoritySnapshotData(peerIds: NonEmptyList[PeerId]): F[MajoritySnapshotData] = {
+        val maybeData =
+          for {
+            peers <- OptionT(getL0Peers(peerIds))
+            majorityOrdinal <- OptionT(getMajorityOrdinal(peers))
+            majorityHash <- OptionT(getMajorityHash(peers, majorityOrdinal))
+          } yield MajoritySnapshotData(majorityOrdinal, majorityHash)
+
+        maybeData.getOrElseF(NoMajoritySnapshotData.raiseError[F, MajoritySnapshotData])
+      }
+
+      private def getMajorityHash(peers: NonEmptyList[L0Peer], ordinal: SnapshotOrdinal): F[Option[Hash]] =
+        peers.toList
+          .parTraverseN(10)(p =>
+            attemptOption(l0GlobalSnapshotClient.getHash(ordinal).run(p), s"Unable to obtain hash for majority peer $p")
+          )
+          .map(_.flatMap(_.flatten))
+          .flatTap(hashes => logger.debug(s"** Majority Hashes $hashes"))
+          .map(selectMajorityHash)
+
+      private def getPeerInTheMajority(ordinal: SnapshotOrdinal, hash: Hash, peerSet: NonEmptySet[L0Peer]): F[L0Peer] =
+        Random
+          .shuffle(List.from(peerSet.toSortedSet))
+          .tailRecM[F, L0Peer] {
+            case Nil => NoPeerAlignedWithMajority.raiseError[F, Either[List[L0Peer], L0Peer]]
+            case head :: tail =>
+              l0GlobalSnapshotClient
+                .getHash(ordinal)
+                .run(head)
+                .map {
+                  case Some(peerHash) if peerHash === hash => head.asRight[List[L0Peer]]
+                  case _                                   => tail.asLeft[L0Peer]
+                }
+          }
+
+      private def getL0Peers(peerIds: NonEmptyList[PeerId]): F[Option[NonEmptyList[L0Peer]]] =
+        peerIds
+          .traverse(globalL0ClusterStorage.getPeer)
+          .map(_.toList.flatten.toNel)
+
+      private def getMajorityOrdinal(peers: NonEmptyList[L0Peer]): F[Option[SnapshotOrdinal]] =
+        peers.toList
+          .parTraverseN(10)(p =>
+            attemptOption(l0GlobalSnapshotClient.getLatestOrdinal.run(p), s"Unable to retrieve ordinal for majority peer $p")
+          )
+          .map(_.flatten)
+          .map(pickMajority[List, SnapshotOrdinal])
+
+      private def selectMajorityHash(hashes: List[Hash]): Option[Hash] = {
+        val minCount = max(minimumMajorityPeerIds, ceil(0.5 * (1 + hashes.size)).toInt)
+        hashes
```

Comment:
I think you can do something like:
```
pickMajority(hashes)
    .filter(h => hashes.count(_ === h) >= minCount)
```
to have it simpler and not copy paste parts of pickMajority functionality.